### PR TITLE
Fix SwiftLint warnings

### PR DESCRIPTION
**Pull request by DanielKamkha**
Resolves: #733
Signed-off-by: Daniel Kamkha <daniel.kamkha@here.com>

### CODE REVIEW

File: functional-tests/functional/swift/Tests/EnumsTests.swift [hunk 1]
```diff
@@ -66,10 +66,13 @@ class EnumsTests: XCTestCase {
     func testCodableWithDeprecated() {
         let value = EnumWithDeprecatedItems.baz
 
-        let data = try! JSONEncoder().encode(value)
-        let result = try! JSONDecoder().decode(EnumWithDeprecatedItems.self, from: data)
-
-        XCTAssertEqual(result, value)
+        let data = try? JSONEncoder().encode(value)
+        if data == nil {
```
Comment (Hsilgos) [inline functional-tests/functional/swift/Tests/EnumsTests.swift:70]: Hm... How about 

```
if let data = try? JSONEncoder().encode(value) {
} else {
  XCTFail(...)
}
```

or 

```
guard let data = try? JSONEncoder().encode(value) else {
  XCTFail(...)
  return // Not sure that it's necessary
}
```

or 
```
let data = try? JSONEncoder().encode(value)
XCTAssertNotNil(data)
if data != nil {
 ...
}
```

File: functional-tests/functional/swift/Tests/SerializationTests.swift [hunk 2]
```diff
@@ -49,25 +49,32 @@ class SerializationTests: XCTestCase {
...
-        let data = try! JSONEncoder().encode(serializableStruct)
-        let resultStruct = try! JSONDecoder().decode(SerializableStruct.self, from: data)
+    func serializableEncodeDecodeRoundTrip() {
+        let serializableStruct = createSerializableStruct()
+        let data = try? JSONEncoder().encode(serializableStruct)
+        if data == nil {
+            XCTAssertNotNil(data)
+            return
+        }
+        let resultStruct = try? JSONDecoder().decode(SerializableStruct.self, from: data!)
```
Comment (Hsilgos) [inline functional-tests/functional/swift/Tests/SerializationTests.swift:61]: Also guards here and above?

File: functional-tests/functional/swift/Tests/EnumsTests.swift (hunk 1 above)
Comment (DanielKamkha) [reply functional-tests/functional/swift/Tests/EnumsTests.swift:70]: Done.

File: functional-tests/functional/swift/Tests/SerializationTests.swift (hunk 2 above)
Comment (DanielKamkha) [reply functional-tests/functional/swift/Tests/SerializationTests.swift:61]: Done.

**Review by Hsilgos — APPROVED** [review]

**Pull request merged**